Assets/HoloToolkit/Input/Scripts/Cursor/Cursor.cs

```diff
@@ -417,10 +417,14 @@ public virtual CursorStateEnum CheckCursorState()
         {
             if (cursorState != CursorStateEnum.Contextual)
             {
-                if (IsInputSourceDown)
+                if (IsInputSourceDown && visibleHandsCount > 0)
                 {
                     return CursorStateEnum.Select;
                 }
+                else if (IsInputSourceDown && visibleHandsCount == 0)
```

Comment:
It looks like taking this branch prevents the cursor from going to `CursorStateEnum.Release` from `Select`.  How about instead if we just reset IsInputSourceDown as part of `OnSourceLost` and left `CheckCursorState` the same?
```cs
        public virtual void OnSourceLost(SourceStateEventData eventData)
        {
            visibleHandsCount--;
            if (visibleHandsCount == 0)
            {
                IsHandVisible = false;
                IsInputSourceDown = false;
            }
        }
```